src/material/form-field/form-field.scss

```diff
@@ -165,3 +165,11 @@
 .mdc-notched-outline--upgraded .mdc-floating-label--float-above {
   max-width: calc(100% * 4 / 3 + 1px);
 }
+
+.mat-mdc-form-field-type-mat-input {
```

Comment:
Is this logic specific to the mat-input? I would think the same label logic should apply regardless of whether its an input, select, etc.

Reply:
Hi @mmalerba,

Yes i checked logic should be applied regardless of type. I have submitted revised changes.